pkg/cloudprovider/provider/kubevirt/provider.go

```diff
@@ -72,16 +72,24 @@ func New(configVarResolver *providerconfig.ConfigVarResolver) cloudprovidertypes
 }
 
 type Config struct {
-	Kubeconfig       string
-	RestConfig       *rest.Config
-	DNSConfig        *corev1.PodDNSConfig
-	DNSPolicy        corev1.DNSPolicy
-	CPUs             string
-	Memory           string
-	Namespace        string
-	SourceURL        string
+	Kubeconfig         string
+	RestConfig         *rest.Config
+	DNSConfig          *corev1.PodDNSConfig
+	DNSPolicy          corev1.DNSPolicy
+	CPUs               string
+	Memory             string
+	Namespace          string
+	OsImage            string
```

Comment:
Just as discussed earlier, I guess we need to make this a bit more generic type so that we can include importing images from url or importing an image via data volume cloning. Thus I would suggest that we convert this type from string to struct. Something like this: 
```go
// URL and DataVolumeName are mutually exclusive 
type OSImage struct{
// if not empty, importing image from an http endpoint 
URL                        string
// if not empty, importing image from a DV clone 
DataVolumeName string
}